pyccel/codegen/printing/ccode.py

```diff
@@ -1649,7 +1652,13 @@ def _print_Allocate(self, expr):
         free_code = ''
         variable = expr.variable
         if isinstance(variable.class_type, (HomogeneousListType, HomogeneousSetType, DictType)):
-            return ''
+            if len(variable.alloc_shape) > 0 and isinstance(variable.class_type, HomogeneousListType):
+                size = self._print(variable.alloc_shape[0])
+                variable_address = self._print(ObjectAddress(expr.variable))
+                container_type = self.get_c_type(expr.variable.class_type)
+                return f'{container_type}_reserve({variable_address}, {size});\n'
+            else:
+                return ''
         if variable.rank > 0:
             #free the array if its already allocated and checking if its not null if the status is unknown
```

Comment:
Should there be a call to clear for lists if the status is unknown?

Reply:
could you elaborate please?

Reply:
The comment at the bottom of the code section says that Deallocate is called for arrays if the status of the Allocate node is 'unknown'. The equivalent function for lists is probably clear. I was wondering if the Allocate status should be used here